x-pack/legacy/plugins/monitoring/server/kibana_monitoring/__tests__/bulk_uploader.js

```diff
@@ -307,5 +312,89 @@ describe('BulkUploader', () => {
         done();
       }, CHECK_DELAY);
     });
+
+    it('uses a direct connection to the monitoring cluster, when configured', done => {
+      const prodClusterUuid = '1sdfd5';
+      const prodCluster = {
+        callWithInternalUser: sinon
+          .stub()
+          .withArgs('monitoring.bulk')
+          .callsFake(arg => {
+            let resolution = null;
+            if (arg === 'info') {
+              resolution = { cluster_uuid: prodClusterUuid };
+            }
+            return new Promise(resolve => resolve(resolution));
+          }),
+      };
+      const monitoringCluster = {
+        callWithInternalUser: sinon
+          .stub()
+          .withArgs('bulk')
+          .callsFake(() => {
+            return new Promise(resolve => setTimeout(resolve, CHECK_DELAY + 1));
+          }),
+      };
+
+      const collectorFetch = sinon.stub().returns({
+        type: 'kibana_stats',
+        result: { type: 'kibana_stats', payload: { testData: 12345 } },
+      });
+
+      const collectors = new MockCollectorSet(server, [
+        {
+          fetch: collectorFetch,
+          isReady: () => true,
+          formatForBulkUpload: result => result,
+          isUsageCollector: false,
+        },
+      ]);
+      const customServer = {
+        ...server,
+        elasticsearchPlugin: {
+          createCluster: () => monitoringCluster,
+          getCluster: name => {
+            if (name === 'admin' || name === 'data') {
+              return prodCluster;
+            }
+            return monitoringCluster;
+          },
+        },
+        config: {
+          get: key => {
+            if (key === 'monitoring.elasticsearch') {
+              return {
+                hosts: ['http://localhost:9200'],
+                username: 'tester',
+                password: 'testing',
+                ssl: {},
+              };
+            }
+            return null;
+          },
+        },
+      };
+      const kbnServerStatus = { toJSON: () => ({ overall: { state: 'green' } }) };
+      const kbnServerVersion = 'master';
+      const uploader = new BulkUploader({
+        ...customServer,
+        interval: FETCH_INTERVAL,
+        kbnServerStatus,
+        kbnServerVersion,
+      });
+      uploader.start(collectors);
+      setTimeout(() => {
+        uploader.stop();
+        const firstCallArgs = monitoringCluster.callWithInternalUser.firstCall.args;
+
+        expect(firstCallArgs[0]).to.be('bulk');
+        expect(firstCallArgs[1].body[0].index._index).to.be(
+          `.monitoring-kibana-${MONITORING_SYSTEM_API_VERSION}-${moment().format('YYYY.MM.DD')}`
```

Comment:
Doing `moment()` might make the tests a bit flaky (in the off chance that the date switch happens between an async operation or any other daylight-savings/tz shenanigans). Any chance to use predefined dates instead?

Reply:
Sure, will fix that up